CarMethod.java

```diff
@@ -0,0 +1,121 @@
+        System.out.println("Sorted list: ");
+        Comparator<Car> compareByPrice =
+                Comparator.comparing(Car::getPrice);
+        carList.stream()
+                .sorted(compareByPrice)
+                .collect(Collectors.toList())
+                .forEach(System.out::println);
+
+        //req 9: matching a condition based on an Integer field
+        System.out.println("AnyMatch ");
+        boolean answer = carList.stream()
+                .anyMatch(x -> x.getBattery() > 1000);
+        System.out.println(answer);
+
+        //req 10: Optional, check with ifPresent
+        System.out.println("Explaining Optional: ");
+        Optional<Boolean> autoPilot = Optional.of(true);
+
+        Boolean answer1 = false;
+        Boolean answer2 = null;
+        autoPilot.ifPresent(System.out::println);
+    }
+
+    // req 4: generate random Strings and find MAX + explain
+    //1. the method generate a set of Strings containing different length;
+    //2. using a natural order to compare, the System will save first element in the set as MAX;
+    //3. the system compare the "false" MAX with the rest of elements;
+    //4. if a biggest number is found, this will be the new value of MAX;
+    //5. the process is finished when no number greater than MAX is founded;
+    static void generateStringsFindTheMax() {
+        Set<String> randomStringSet = new HashSet<>(5);
+        while (5 > randomStringSet.size()) randomStringSet.add(getString());
+        System.out.println(randomStringSet);
+        System.out.println(randomStringSet.stream().max(String::compareTo));
+        
+        //req 11: fastest way to find the shortest String
+        System.out.println(randomStringSet.stream().min(String::compareTo));
+    }
+    private static String getString() {
+        StringBuilder builder = new StringBuilder();
+        int character = (int) (Math.random() * ALPHABETICAL_STRING.length());
+        for (int i = 0; i <= character; i++) {
+            builder.append(ALPHABETICAL_STRING.charAt(character));
+        }
+        return builder.toString();
+    }
+
+    // req 5: count number of integer and mapping
+    public void generateIntegersCountAndMap() {
+        List<Integer> integerList = Arrays.asList(57, 23, 44, 97);
+        System.out.println((long) integerList.size());
+        integerList.stream()
+                .filter(Objects::nonNull)
+                .map(x -> integerList.get(1) * Math.sqrt(2.0))
+                .forEach(System.out::println);
+    }
+
+    // req 6: create a map and sort the elements
+    public void sortAMap() {
+        HashMap<String, Integer> integerHashMap = new HashMap<>();
+        integerHashMap.put("start", 100);
+        integerHashMap.put("drive", 85);
+        integerHashMap.put("stop", 3);
+
+        long count = integerHashMap.values()
+                .stream()
+                .filter(x -> x > 10)
+                .count();
+        System.out.println(count);

```

Comment:
please try integrating the LOGGER for next time since Syso is rarely used in applications: https://www.vogella.com/tutorials/Logging/article.html